packages/editor/src/components/rich-text/input-event.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+export class RichTextInputEvent extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onInput = this.onInput.bind( this );
+	}
+
+	onInput( event ) {
+		if ( event.inputType === this.props.inputType ) {
+			this.props.onInput();
```

Comment:
Is there a need to `preventDefault` as well, to ensure it's fully controlled?

Reply:
There's not really anything to prevent. The `input` event happens after the fact.

Reply:
> There's not really anything to prevent. The `input` event happens after the fact.

And to that point, an `input` event is in-fact not cancelable.

https://w3c.github.io/uievents/#event-type-input